w1r3/go/main.go

```diff
@@ -390,43 +370,106 @@ func downloadStep(ctx context.Context,
 			downloadSpan.RecordError(err)
 			downloadSpan.End()
 		}
-		cpuStart, err := cpuStart()
+		start, err := StartOp()
 		if err != nil {
-			endSpan("error during cpuStart()", err)
+			endSpan("error during StartOp()", err)
 			continue
 		}
-		latencyStart := time.Now()
 		objectReader, err := objectHandle.NewReader(downloadContext)
 		if err != nil {
 			endSpan("error while opening reader", err)
 			continue
 		}
+		// Only record data in the histogram for successful downloads. Otherwise
+		// we are mixing results for different download sizes.
 		if _, err := io.Copy(io.Discard, objectReader); err != nil {
 			endSpan("error while closing reader", err)
 			continue
 		}
-		// Only record data in the histogram for successful downloads. Otherwise
-		// we are mixing results
-		downloadSpan.End()
-		latencyDuration := time.Since(latencyStart)
-
-		cpuDuration, err := cpuElapsed(cpuStart)
+		err = RecordOp(downloadContext, config, stepConfig, objectSize, &start, op)
 		if err != nil {
-			endSpan("error during cpuEnd()", err)
+			endSpan("error during RecordOp()", err)
 			continue
 		}
-		cpuNanosPerByte := float64(cpuDuration.Nanoseconds()) / float64(objectSize)
-		stepConfig.cpu.Record(downloadContext, cpuNanosPerByte, metric.WithAttributes(
-			append([]attribute.KeyValue{attribute.String("ssb.op", op)},
-				stepConfig.commonAttributes...)...))
-		stepConfig.latency.Record(downloadContext, latencyDuration.Seconds(),
-			metric.WithAttributes(
-				append([]attribute.KeyValue{attribute.String("ssb.op", op)},
-					stepConfig.commonAttributes...)...))
+		downloadSpan.End()
+	}
+}
+
+type Usage struct {
+	start time.Time
+	cpu   time.Duration
+	mem   uint64
+}
+
+func StartOp() (Usage, error) {
```

Comment:
This and RecordOp seem like they should just be methods on Usage with a pointer receiver maybe? Seems like it would be easier to follow.

Reply:
`StartOp()` looks constructor-like to me, so I left this as-is.  Agreed on `RecordOp()`.